ngraph/core/src/op/transpose.cpp

```diff
@@ -82,65 +83,53 @@ shared_ptr<Node> op::v1::Transpose::clone_with_new_inputs(const OutputVector& ne
 namespace
 {
     template <element::Type_t ET>
-    std::vector<int64_t> get_vector(const HostTensorPtr& arg)
+    bool evaluate(const HostTensorPtr& arg1, const HostTensorPtr& arg2, const HostTensorPtr& out)
     {
-        std::vector<int64_t> rc;
-        auto p = arg->get_data_ptr<ET>();
-        for (size_t i = 0; i < shape_size(arg->get_shape()); i++)
+        const Shape& data_shape = arg1->get_shape();
+        const Shape& axis_shape = arg2->get_shape();
+        Shape out_shape(data_shape.size());
+        auto axis_order = arg2->get_data_ptr<ET>();
+        for (size_t i = 0; i < axis_shape[0]; ++i)
         {
-            rc.push_back(p[i]);
+            out_shape[i] = data_shape[axis_order[i]];
         }
-        return rc;
+        out->set_shape(out_shape);
+        runtime::reference::transpose(arg1->get_data_ptr<char>(),
+                                      data_shape,
+                                      arg1->get_element_type().size(),
+                                      axis_order,
+                                      axis_shape,
+                                      out->get_data_ptr<char>());
+        return true;
     }
 
     bool evaluate_transpose(const HostTensorPtr& arg1,
                             const HostTensorPtr& arg2,
                             const HostTensorPtr& out)
     {
         element::Type_t axis_type = arg2->get_element_type();
-
-        std::vector<int64_t> axis_order;
+        bool rc = true;
         switch (axis_type)
         {
-        case element::Type_t::i8: axis_order = get_vector<element::Type_t::i8>(arg2); break;
-
-        case element::Type_t::i16: axis_order = get_vector<element::Type_t::i16>(arg2); break;
-
-        case element::Type_t::i32: axis_order = get_vector<element::Type_t::i32>(arg2); break;
-
-        case element::Type_t::i64: axis_order = get_vector<element::Type_t::i64>(arg2); break;
-
-        case element::Type_t::u8: axis_order = get_vector<element::Type_t::u8>(arg2); break;
-
-        case element::Type_t::u16: axis_order = get_vector<element::Type_t::u16>(arg2); break;
-
-        case element::Type_t::u32: axis_order = get_vector<element::Type_t::u32>(arg2); break;
-
-        case element::Type_t::u64: axis_order = get_vector<element::Type_t::u64>(arg2); break;
-
-        default: throw ngraph_error("axis element type is not integral data type");
+            TYPE_CASE(i8)(arg1, arg2, out);
```

Comment:
You can use the old code if you remove template argument from transpose reference implementation.

Reply:
see comment above

Reply:
transpose related changes were reverted